### ISSUE TITLE
TestTest/custom_condition_inputs fails for 1.6.6

### ISSUE THREAD

**Issue by hashworks** [hashicorp/terraform#34421]
### Terraform Version

```shell
1.6.6
```

### Terraform Configuration Files

None, test fails

### Debug Output

```
--- FAIL: TestTest (0.21s)
    --- FAIL: TestTest/custom_condition_inputs (0.00s)
        test_test.go:250: output didn't contain expected string:
            
            main.tftest.hcl... in progress
              run "test"... fail
            main.tftest.hcl... tearing down
            main.tftest.hcl... fail
            
            Failure! 0 passed, 1 failed╷
            │ Error: Invalid value for variable
            │ 
            │   on main.tf line 2:
            │    2: variable "input" {
            │     ├────────────────
            │     │ var.input is "some value"
            │ 
            │ this should definitely fail
            │ 
            │ This was checked by the validation rule at main.tf:5,3-13.
            ╵
            .
FAIL
```

### Expected Behavior

All tests are green.

### Actual Behavior

Test `TestTest/custom_condition_inputs` fails.

### Steps to Reproduce

```
git checkout v1.6.6
go mod download
go build
go test -mod=readonly ./...
```

### Additional Context

Golang version 1.21.5.

### References

- #34410

**Comment by liamcervante** [comment]
This was fixed in main in https://github.com/hashicorp/terraform/pull/34320.

@hashworks, does this need to be backported for you? Note, that we won't be able to update the 1.6.6 tag even if we backport this but maybe it helps you even if it's just in the 1.6 branch?

**Comment by hashworks** [comment]
@liamcervante Nah, if it's fixed it's fine. I have the principle that I ignore failing tests as long as they are non-critical and known to upstream.

https://gitlab.archlinux.org/archlinux/packaging/packages/terraform/-/blob/main/PKGBUILD?ref_type=heads#L47

Can I expect the fix in >=1.7.0 ?

**Comment by liamcervante** [comment]
> Can I expect the fix in >=1.7.0 ?

Yes, I think you can verify that via the latest release candidate:

```
git checkout v1.7.0-rc1
go mod download
go build
go test -mod=readonly ./...
```

Thanks! I will close this issue.
